src/systemctl/systemctl-start-unit.c

```diff
@@ -348,7 +348,18 @@ int verb_start(int argc, char *argv[], void *userdata) {
         }
 
         if (arg_wait) {
-                r = bus_call_method_async(bus, NULL, bus_systemd_mgr, "Subscribe", NULL, NULL, NULL);
+                bool new_signal_available = false;
+
+                r = bus_jobremoved2_signal_available(bus);
```

Comment:
this means the Subscribe() call must come first, and then match installation to JobRemovedEx second, but that should not matter.

Reply:
But there are some places, like in timedated, where it just watches for the signal, it doesn't subscribe. So the only thing I can do in that situation is instrospection. What's the issue with introspection? It's cached, so even if it's a bit slow, it only happens on first invocation.

Reply:
it's actually a bug if we don't subscribe. We really should. 

maybe add a helper for all this. i.e. bus_subscribe_job_removed() or so, which will do the right thing, and then change all code to just call that.

Reply:
Two more issues with that:

1) in all places where it happens, the subscribe is done as the last thing, after all signals match are installed, I assume to avoid races - this would have to be swapped
2) in all places where it happens, the subscribe is called async, so there's no immediate feedback on whether the method exists or not, these would have to all become sync, blocking calls

Reply:
implemented this - somewhat - as usage varies too wildly